Battle.js

```diff
@@ -0,0 +1,124 @@
+/*TODO: 
+Main battle screen, should display 2 knights with health bars and a 
+button for each different ability*/
+
+import { StyleSheet, Text, View, Button, Image, Dimensions, Pressable } from 'react-native';
+import React, { useState, useRef, useEffect } from 'react';
+import { HealthBar } from "./HealthBar.js"
+import { score } from "./Dodge.js"
+import { useFocusEffect } from '@react-navigation/native';
+import { hookstate, useHookstate, createState } from '@hookstate/core';
+// .value: Gets the value of the state
+// .set(): Sets the value of the state
+
+
+const MAXwidth = Dimensions.get('window').width;
+const MAXheight = Dimensions.get('window').height;
+//const [health, setHealth] = useState(100);
+const health2 = createState(100);
+
+export default function Battle({ navigation, route }) {
+    const healthState2 = useHookstate(health2);
+    //function which runs when the screen is 'focused' e.g. visible
+    useFocusEffect(
+        React.useCallback(() => {
+            healthState2.set(calculateDamage())
+        }, [])
+    );
+    console.log(healthState2.value)
+
+    function calculateDamage() {
+        console.log(score)
+        let newHealth = healthState2.value - (score * .15)
+    return (newHealth)
+    }
+
+
+    return (
+        <View style={{
+            width: MAXwidth, height: MAXheight, backgroundColor: '#404040',
+            position: 'relative',
+        }}>
+
+            <View style={{
+                width: MAXwidth / 2, height: MAXheight,
+                left: 100, top: 0
+            }}>
+                <HealthBar playerNum={1} health={100} />
+
+                <Image
```

Comment:
Since you use it twice, it's not a bad idea to make a class (or function) that does the common logic for you.

i.e.
```
function KnightImage(props) {
  return (<Image
     source={require("./assets/knight1.jpg")}
     style={{
        ...
        width: props.MAXwidth / 10
        ...
      }}
  />)
}
```